core/src/main/scala/kafka/server/DelegationTokenManagerZk.scala

```diff
@@ -178,6 +178,12 @@ class DelegationTokenManagerZk(config: KafkaConfig,
 
         val issueTimeStamp = time.milliseconds
         val maxLifeTime = if (maxLifeTimeMs <= 0) tokenMaxLifetime else Math.min(maxLifeTimeMs, tokenMaxLifetime)
+
+        if (checkTimestampOverflow(issueTimeStamp, maxLifeTime)) {
+          responseCallback(CreateTokenResult(owner, tokenRequester, -1, -1, -1, "", Array[Byte](), Errors.INVALID_TIMESTAMP))
```

Comment:
Could we set upper bound `Long.MAX_VALUE` instead of throwing exception? That makes `Long.MAX_VALUE` valid in request. WDYT?

Reply:
Make sense, this change will make api easier to use.